src/schematools/provenance/create.py

```diff
@@ -41,21 +41,16 @@ def set_dataset_for_final_listing(self, dataschema: DatasetSchema):
                 if item == "tables":
                     self.get_provenance_per_table(dataschema[item])
 
-    def get_provenance_per_table(self, dictionary):
+    def get_provenance_per_table(self, dicts):
         """Calling the processing to extract the provenance data per table"""
 
         # loop trough all existing tables and proces data for provenance
         # on each and add them each to the final_dict outcome
-        for n in range(0, self.number_of_tables):
-
-            self.temp_dic_tables["table"] = dictionary[n]["id"]
-            self.temp_dic_tables["provenance"] = (
-                dictionary[n]["provenance"]
-                if "provenance" in dictionary[n]
-                else "na"  # not applicable
-            )
+        for d in dicts[: self.number_of_tables]:
+            self.temp_dic_tables["table"] = d["id"]
+            self.temp_dic_tables["provenance"] = d.get("provenance", "na")
```

Comment:
had je nog gekeken waar die `na` gebruikt wordt? lijkt me haast ook zo'n onhandige waarde.

Reply:
Nee, ik heb alleen gekeken of de tests blijven werken.